content/300-guides/100-performance-and-optimization/150-connection-management/index.mdx

```diff
@@ -0,0 +1,306 @@
+
+#### Optimizing for parallel requests
+
+If you rarely or never exceed the database connection limit with the pool size set to 1, you can further optimize the connection pool size. Consider a function that is sends queries in parallel:
+
+```ts
+Promise.all() {
+  query1,
+  query2,
+  query3
+  query4,
+  ...
+}
+```
+
+If the `connection_limit` is 1, this function is forced to send queries **serially** (one after the other) rather than **in parallel**. This slows down the function's ability to process requests, and may result in pool timeout errors. Tune the `connection_limit` parameter until a traffic spike:
+
+* Does not exhaust the database connection limit
+* Does not result in pool timeout errors
+
+### <inlinecode>PrismaClient</inlinecode> in serverless environments
+
+#### Instantiate `PrismaClient` outside the handler
+
+Instantiate `PrismaClient` [outside the scope of the function handler](https://github.com/prisma/e2e-tests/blob/5d1041d3f19245d3d237d959eca94d1d796e3a52/platforms/serverless-lambda/index.ts#L3) to increase the chances of reuse. As long as the handler remains 'warm' (in use), the connection is potentially reusable:
+
+```ts highlight=3;normal
+import { PrismaClient, Prisma } from '@prisma/client'
+
+const client = new PrismaClient()
+
+export async function handler() {
+    /* ... */
+}
+```
+
+#### Do not explicitly <inlinecode>$disconnect()</inlinecode>
+
+You [do not need to explicitly `$disconnect()`](../../../concepts/components/prisma-client/working-with-prismaclient/connection-management#calling-disconnect-explicitly) <span class="concept"></span> at the end of a function, as there is a possibility that the container might be reused. Opening a new connection takes time and slows down your function's ability to process requests.
+
+### Other serverless considerations
+
+#### Container reuse
+
+There is no guarnatee that subsequent nearby invocations of a function will hit the same container - for example, AWS can choose to create a new container at any time.
+
+Code should assume the container to be stateless and create a connection only if it does not exist - Prisma Client JS already implements this logic.
+
+#### Zombie connections
+
+Containers that are marked "to be removed" and are not being reused still **keep a connection open** and can stay in that state for some time (unknown and not documented from AWS). This can lead to sub-optimal utilization of the database connections.
+
+A potential solution is to **clean up idle connections** ([`serverless-mysql`](https://github.com/jeremydaly/serverless-mysql) implements this idea).
+
+> **Note**: This is a suggested solution, not an official recommendation. Refer to the documentation for your chosen serverless platform for official recommendations.
```

Comment:
Problem: `serverless-mysql` is not usable with Prisma, but only with the normal `mysql` node package. So the Note does not really make sense here unfortunately - people _can_ not use this suggested solution.

(We might have an equivalent for this soon-ish, but not yet.)